### PR TITLE
enhance: New Picker no need `needConfirm` when `multiple`

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

* No need `needConfirm` when `multiple`
* Fix min & max Date limitation switch bug

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     不用写在 Changelog 中      |
| 🇨🇳 Chinese |     不用写在 Changelog 中      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by li-jia-nan — DISMISSED** [review]

File: components/date-picker/index.zh-CN.md [hunk 1]
```diff
@@ -113,7 +113,7 @@ import 'dayjs/locale/zh-cn';
 | minDate | 最小日期，同样会限制面板的切换范围 | dayjs | - | 5.14.0 |
 | maxDate | 最大日期，同样会限制面板的切换范围 | dayjs | - | 5.14.0 |
 | mode | 日期面板的状态（[设置后无法选择年份/月份？](/docs/react/faq#当我指定了-datepickerrangepicker-的-mode-属性后点击后无法选择年份月份)） | `time` \| `date` \| `month` \| `year` \| `decade` | - |  |
-| needConfirm | 是否需要确认按钮，为 `false` 时失去焦点即代表选择 | boolean | - | 5.14.0 |
+| needConfirm | 是否需要确认按钮，为 `false` 时失去焦点即代表选择。当设置 `multiple` 时默认为 `false` | boolean | - | 5.14.0 |
```
Comment (yoyo837) [inline components/date-picker/index.zh-CN.md:116]: `needConfirm` 本身的默认值是 `true`?

Comment (yoyo837) [reply components/date-picker/index.zh-CN.md:116]: 如果本身是 `true`, 列 `Default` 没写。

Comment (zombieJ) [reply components/date-picker/index.zh-CN.md:116]: 不是，根据 `picker` 属性动态变化的。

Comment (yoyo837) [reply components/date-picker/index.zh-CN.md:116]: 那是不是应该列一下分别是哪些情况为true？

**Comment by li-jia-nan** [comment]
大佬，这个可以合了吗？

**Review by MadCcc — APPROVED** [review]

**Pull request merged**